src/commands/lambda/flare.ts

```diff
@@ -40,105 +49,170 @@ export class LambdaFlareCommand extends Command {
     }
+    assert(this.functionName !== undefined)
 
     // Get AWS credentials
     this.context.stdout.write('\n🔑 Getting AWS credentials...\n')
     try {
       this.credentials = await getAWSCredentials()
     } catch (err) {
-      this.context.stderr.write(commonRenderer.renderError(err))
+      this.context.stderr.write(commonRenderer.renderError(err.message))
 
       return 1
     }
     if (this.credentials === undefined) {
-      this.context.stdout.write(commonRenderer.renderNoAWSCredentialsFound())
+      this.context.stdout.write('\n' + commonRenderer.renderNoAWSCredentialsFound())
       try {
         await requestAWSCredentials()
       } catch (err) {
-        this.context.stderr.write(commonRenderer.renderError(err))
+        this.context.stderr.write(commonRenderer.renderError(err.message))
 
         return 1
       }
     }
 
-    // Get Lambda function configuration
-    this.context.stdout.write('\n🔍 Getting Lambda function configuration...\n')
+    // Get and print Lambda function configuration
+    this.context.stdout.write('\n🔍 Fetching Lambda function configuration...\n')
     const lambdaClientConfig: LambdaClientConfig = {
       region,
       credentials: this.credentials,
     }
     const lambdaClient = new LambdaClient(lambdaClientConfig)
     let config
     try {
-      config = await getLambdaFunctionConfig(lambdaClient, this.functionName ?? '')
+      config = await getLambdaFunctionConfig(lambdaClient, this.functionName)
     } catch (err) {
       this.context.stderr.write(
         commonRenderer.renderError(`Unable to get Lambda function configuration: ${err.message}`)
       )
 
       return 1
     }
+    const configStr = util.inspect(config, false, undefined, true)
+    this.context.stdout.write(`\n${configStr}\n`)
 
-    // Print config
-    const configStrColored = util.inspect(config, false, undefined, true)
-    this.context.stdout.write(`\n${configStrColored}\n`)
-    if (this.isDryRun) {
-      this.context.stdout.write('\n🚫 The configuration was not sent as it was executed in dry run mode.\n')
+    // Get CloudWatch logs
+    const logs: [string, OutputLogEvent[]][] = []
+    if (this.withLogs) {
+      this.context.stdout.write('\n☁️ Getting CloudWatch logs...\n')
+      const cwlClient = new CloudWatchLogsClient({region})
+      const functionName = this.functionName.startsWith('arn:aws') ? this.functionName.split(':')[6] : this.functionName
+      const logGroupName = `/aws/lambda/${functionName}`
+      let logStreamNames
+      try {
+        logStreamNames = await getLogStreamNames(cwlClient, logGroupName)
+      } catch (err) {
+        this.context.stderr.write(commonRenderer.renderError(`Unable to get log streams: ${err.message}`))
+
+        return 1
+      }
+      if (logStreamNames === undefined) {
+        this.context.stdout.write(
+          commonRenderer.renderSoftWarning('No CloudWatch streams were found. Logs will not be retrieved or sent.\n')
+        )
+      } else {
```

Comment:
Avoid using `else`

Reply:
Why should I avoid using `else` in this scenario?
I'm also not sure how to achieve the same logic without using `else` and without adding additional complexity.

Reply:
As we talked about, I have broken this down into its own function `getAllLogs`.
Usage: `logs = await getAllLogs(region!, this.functionName)`

What do you think of these changes?